### PR TITLE
 Closes #7142: Sanitize url in HttpIconLoader 

### LINKED ISSUE #7142: Crash IllegalArgumentException: Unsupported URI scheme when downloading icon
Originally reported here: https://github.com/mozilla-mobile/fenix/issues/10880

## Steps to reproduce
Visit https://www.gruppoapi.com/ricerca-stazioni-servizio

> 2020-05-27 20:00:14.677 9383-9590/org.mozilla.fenix.debug D/HttpIconLoader: IllegalArgumentException while trying to download icon resource https://www.gruppoapi.com/ricerca-stazioni-servizio/images/logo-gruppoapi-shared.png
    
    java.lang.IllegalArgumentException: Unsupported URI scheme
        at org.mozilla.geckoview.GeckoWebExecutor.fetch(GeckoWebExecutor.java:128)
        at mozilla.components.browser.engine.gecko.fetch.GeckoViewFetchClient.fetch(GeckoViewFetchClient.kt:64)
        at mozilla.components.browser.icons.loader.HttpIconLoader.load(HttpIconLoader.kt:54)
        at mozilla.components.browser.icons.BrowserIconsKt.load(BrowserIcons.kt:276)
        at mozilla.components.browser.icons.BrowserIconsKt.access$load(BrowserIcons.kt:1)
        at mozilla.components.browser.icons.BrowserIcons.loadIconInternal(BrowserIcons.kt:127)
        at mozilla.components.browser.icons.BrowserIcons.access$loadIconInternal(BrowserIcons.kt:77)
        at mozilla.components.browser.icons.BrowserIcons$loadIcon$1.invokeSuspend(BrowserIcons.kt:110)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:56)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:919)

I confirmed catching the `IllegalArgumentException` in `HttpIconLoader` fixes this, which we should probably do because crashing on fetching icons seems silly, but I thought this was interesting to bring up before catching it because it's unclear to me why this website would be an `Unsupported URI scheme`.

### PR DESCRIPTION
**Pull request by Amejia481**
---

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

**Comment by Amejia481** [comment]
@pocmo I preferred to not add a `IllegalArgumentException` to the exception list on [HttpIconLoader.load]( https://github.com/mozilla-mobile/android-components/blob/ff3d7a81cad7c79badae41576696b547f62a2d8f/components/browser/icons/src/main/java/mozilla/components/browser/icons/loader/HttpIconLoader.kt#L60) as it is a implementation detail of [GeckoViewFetch/GeckoWebExecutor](https://dxr.mozilla.org/mozilla-central/source/mobile/android/geckoview/src/main/java/org/mozilla/geckoview/GeckoWebExecutor.java#127), that is not necessary true for all implementation of `Client`. What do you think?

**Comment by Amejia481** [comment]
@pocmo 👍  I updated the pr. Now we are sanitizing in both levels `HttpIconLoader` and `GeckoViewFetch`

File: components/browser/icons/src/main/java/mozilla/components/browser/icons/loader/HttpIconLoader.kt [hunk 1]
```diff
@@ -41,7 +41,7 @@ class HttpIconLoader(
         // recently: https://github.com/mozilla-mobile/android-components/issues/2591
 
         val downloadRequest = Request(
-            url = resource.url,
+            url = resource.url.trim(),
```
Comment (pocmo) [inline components/browser/icons/src/main/java/mozilla/components/browser/icons/loader/HttpIconLoader.kt:44]: We can go up higher:
Here we retrieve the values from the webextensions and I think that is the place where we may want to sanitize etc.:
https://github.com/mozilla-mobile/android-components/blob/master/components/browser/icons/src/main/java/mozilla/components/browser/icons/extension/IconMessage.kt#L82-L96

I think we should **not** do it in any of the `Client` implementations and let them continue fail if we pass invalid URLs to them.

Comment (Amejia481) [reply components/browser/icons/src/main/java/mozilla/components/browser/icons/loader/HttpIconLoader.kt:44]: 👍

**Comment by Amejia481** [comment]
I updated the pr :)

**Comment by Amejia481** [comment]
bors retry

**Pull request merged**